app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/mbentity/MBEntityType.java

```diff
@@ -1,25 +1,31 @@
 package org.metabrainz.mobile.data.sources.api.entities.mbentity;
 
 import org.metabrainz.mobile.presentation.features.artist.ArtistActivity;
+import org.metabrainz.mobile.presentation.features.artist.ArtistViewModel;
 import org.metabrainz.mobile.presentation.features.label.LabelActivity;
+import org.metabrainz.mobile.presentation.features.label.LabelViewModel;
 import org.metabrainz.mobile.presentation.features.recording.RecordingActivity;
+import org.metabrainz.mobile.presentation.features.recording.RecordingViewModel;
 import org.metabrainz.mobile.presentation.features.release.ReleaseActivity;
+import org.metabrainz.mobile.presentation.features.release.ReleaseViewModel;
 import org.metabrainz.mobile.presentation.features.release_group.ReleaseGroupActivity;
 
 public enum MBEntityType {
-    ARTIST("artist", ArtistActivity.class),
-    RELEASE("release", ReleaseActivity.class),
-    LABEL("label", LabelActivity.class),
-    RELEASE_GROUP("release-group", ReleaseGroupActivity.class),
-    RECORDING("recording", RecordingActivity.class),
-    EVENT("event", ArtistActivity.class),
-    INSTRUMENT("instrument", ArtistActivity.class);
+    ARTIST("artist", ArtistActivity.class, ArtistViewModel.class),
+    RELEASE("release", ReleaseActivity.class, ReleaseViewModel.class),
+    LABEL("label", LabelActivity.class, LabelViewModel.class),
+    RELEASE_GROUP("release-group", ReleaseGroupActivity.class, ReleaseViewModel.class),
+    RECORDING("recording", RecordingActivity.class, RecordingViewModel.class),
+    EVENT("event", ArtistActivity.class, ArtistViewModel.class),
+    INSTRUMENT("instrument", ArtistActivity.class, ArtistViewModel.class);
 
     public final String name;
     public final Class typeActivityClass;
+    public final Class typeViewModelClass;
 
-    MBEntityType(String entity, Class typeActivityClass) {
+    MBEntityType(String entity, Class typeActivityClass, Class typeViewModelClass) {
```

Comment:
This is kind of weird: the data layer knows about the presentation layer (Android activities and viewmodels). I see it's mostly used by presentation layer and one usage in data layer (`CollectionUtils`). Be careful with this in the future, as this breaks the clean code ideals ⚠️

Reply:
I agree with you. But MBEntityType is not exactly a data class. It is a utility class to concisely maintain a list of supported entities. If I remember correctly, it is nowhere used in the data. I think MBEntityType should actually be moved to the presentation layer.